src/server/graph-generation/interaction/index.js

```diff
@@ -0,0 +1,170 @@
+const _ = require('lodash');
+const fetch = require('node-fetch');
+const pc = require('../../pathway-commons');
+const geneValidator = require('../../enrichment/validation').validatorGconvert;
+const logger = require('./../../logger');
+
+function edgeType(type) {
+  switch(type){
+    case 'in-complex-with':
+    case 'interacts-with':
+      return 'Binding';
+    case 'controls-phosphorylation-of':
+      return 'Phosphorylation';
+    case 'controls-expression-of':
+      return 'Expression';
+    default:
+      return '';
+  }
+}
+
+function getInteractionInfoFromPC(sources) {
+  const idArray = _.uniq(_.concat([], sources)); //convert sources to array
+
+  const geneIds = idArray.map(source =>
+    source.includes('pathwaycommons') ? getGeneIdFromPC(source) : source.replace(/\//g,' ')
+  );
+
+  return getGeneInfoFromNcbi(geneIds).then(result => {
+    const geneResults = result.result;
+    let hgncIds=[];
+    let comments=[];
+    if(!result.esummaryresult) {
+      comments=_.flatten(geneResults.uids.map(gene=>{
+        hgncIds.push(geneResults[gene].name);
+        return _.compact([
+          'Nomenclature Name: '+geneResults[gene].nomenclaturename,
+          'Other Aliases: '+geneResults[gene].name + (geneResults[gene].otheraliases ? ', '+geneResults[gene].otheraliases:''),
+          geneResults[gene].summary && 'Function: '+geneResults[gene].summary
+        ]);
+      }));
+    }
+
+    const metaData = {
+      networkMetadata: {
+        name : hgncIds.length === geneIds.length ? (hgncIds+' Interactions'):' Interactions',
+        datasource : 'Pathway Commons',
+        comments : comments
+      },
+      ids : hgncIds
+    };
+
+    return getInteractionGraphFromPC(hgncIds).then(network => {
+      return {
+        network : network,
+        metaData : metaData
+      };
+    }).catch((e)=>{
+      logger.error(e);
+      return 'ERROR : could not retrieve graph from PC';
+    });
+  });
+}
+
+function getGeneIdFromPC(source) {
```

Comment:
Strip `getGeneIdFromPC` as part of removing `metaData.networkMetadata`. This is fragile anyways (traverse and string subsettting breaks for many cases).